tools/cmd/report.go

```diff
@@ -19,13 +19,12 @@ func newReportCommand() *cobra.Command {
 		Use:   "report",
 		Short: "Generate the weekly activity report",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			theStats, err := stats.New(daysBack, staleMonths, orgName, repoName, devMode,
+			query := util.NewPullRequestQuery(
+				daysBack, staleMonths, orgName, repoName, devMode,
 				util.NewGithubClientSource(configSettings.Github.Token))
-			if err != nil {
-				return errors.Wrap(err, "Could not create stats")
-			}
 
-			err = theStats.IteratePullRequests()
+			theStats := stats.New(query)
+			err := query.IteratePullRequests(theStats.Process)
```

Comment:
It feels wrong to have to pass the query to `stats.New()` and then pass the method to `IteratePullRequests`. Can't the stats type pull the data it needs from the query itself? Or can we avoid storing the query in the stats type?

Reply:
Good call. I've made the interface simply `theStats := stats.Generate(query)` now